### DISCUSSION TITLE
Welcome to DataFrame Discussions!

### DISCUSSION THREAD

**Discussion by hosseinmoein** [hosseinmoein/DataFrame#189] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mdmte50** [comment]
Hi - I used to be a C/C++ whiz 40 years ago, but after 40 years of being away, I'm finding myself stumbling around with the modern tools and the updates that have been implemented over the past 4 decades.  So please excuse the green-ness coming through in my questions below...

Development Environment:
Visual Studio 2022 Pro,
C/C++ Language Standard:  Preview-Features from the Latest C++ Working Draft (/std:c++latest)   // I selected this version instead of ISO C++20 because "zip" is called in the library, but it was dropped from the std library in Visual Studio's C/C++ somewhere in C/C++ 2017 and picked up again circa 2020.  There's only ISO C++2014, 2017, 2020, and the preview version in my Visual Studio 2022 configuration. //

I've got this DataFrame library built on my development machine in Visual Studio 2022, and I was able to properly configure Visual Studio to look at the right lib and include directories.  However, when I #include <DataFrame.h> in my c++ code, I get this error message:

E3251	requires-clause incompatible with variable template "std::ranges::_Compile_time_max_size" (declared at line 58 of "C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\include\ranges")	VinhDataAnalysisDefault	C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\include\ranges	61		

I've spent several hours looking on Google, but I couldn't find a solution to fix the error, but no luck.

Any advice or tips on how to fix the above error?

Thanks in advance for advice and replies from more skilled developers here in this forum.
-Vinh

**Reply by hosseinmoein** [reply]
Hi,

You don't say where this error come from in the DataFrame codebase. It looks like it is related to concepts. 
I cannot reproduce it. Are you using the code in master branch?

**Reply by hosseinmoein** [reply]
I found this also https://developercommunity.visualstudio.com/t/MSVC-ranges-wont-compile-under-std:c/10532125?sort=newest

**Reply by mdmte50** [reply]
Thanks for your reply!  This is so awesome to get a response from the developer so quickly!!!

The error was in the ranges file in the MSVC version 14.39.33519 folder of the Visual Studio installation folder:  (C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.39.33519\include\ranges)

My own code (just to test the DataFrame library) doesn't contain anything other than the standard libraries, DataFrame.h, and a line to display "Hello World!".  I've pasted a copy below.  The error only results when I include the DataFrame library.
Thanks again!

===========================
#include <iostream>
#include <stdlib.h>
#include <conio.h>

#include <DataFrame/DataFrame.h> // There's an error in this library for now...  2024-02-28

using namespace std;

int main()
{
    cout << "Hello World!\n";
}
=============================

**Reply by hosseinmoein** [reply]
First take out `#include <conio.h>` and see if that makes any diff.
If not, it seems like a compiler problem. Also the link I posted above suggests that. Either change the compiler (I am not familiar with MS development environment) or try using `Conan` or `VCPKG` to install DataFrame and make your stuff.

Do examples in DataFrame repo build without a problem?

**Comment by mdmte50** [comment]
I took out #include <conio.h>, but Visual Studio gave me the same build error.
I couldn't get the example in the repo to work (although the build for the repo library itself was successful).
Thanks again for trying to help.
I'm guessing it's a compiler problem based also.  I'll give try Conan and a different compiler tomorrow.

**Reply by mdmte50** [reply]
Great News! I installed the CLANG toolset and had my Visual Studio IDE use that instead of its Microsoft toolset for my project, and the build completed successfully!  No more compile or build errors with the DataFrame library included in my code...  Now I can start playing with the library itself.  Thank you so much for developing the library and for trying to help me yesterday...  your last tip helped me resolve my errors.
